bin/export-db-to-sqlite.sh

```diff
@@ -7,125 +7,305 @@
-# except for the search indices
+# We want to use all the data from the JSON, so let's drop the rows
+# that have been automatically populated during migrate, including all the Wagtail
+# ones, except for wagtailsearch's tables because there's a virtual table that
+# causes fatal problems when loading data it if it's empty. We'll update this later
+# in this script
 
-for tbl in $(sqlite3 $output_db ".tables 'wagtail%'")
+for table in $(sqlite3 $output_db ".tables 'wagtail%'")
 do
-    if [[ $tbl != wagtailsearch_* ]]; then
-        sqlite3 $output_db "DELETE FROM $tbl;"
+    if [[ $table != wagtailsearch_* ]]; then
+        sqlite3 $output_db "DELETE FROM $table;"
+        echo "Purged default-added data from: $table"
     fi
 done
 
-sqlite3 $output_db "DELETE FROM django_content_type";
-sqlite3 $output_db "DELETE FROM django_session";
-sqlite3 $output_db "DELETE FROM auth_group_permissions";
-sqlite3 $output_db "DELETE FROM auth_group";
-sqlite3 $output_db "DELETE FROM auth_permission";
-sqlite3 $output_db "DELETE FROM auth_user_groups";
-sqlite3 $output_db "DELETE FROM auth_user_user_permissions";
-sqlite3 $output_db "DELETE FROM product_details_productdetailsfile";
+# Let's also purge the the Django ones we lined up for cleaning earlier
+for table in "${tables_to_wipe_after_initial_migrate[@]}"
+do
+    sqlite3 $output_db "DELETE FROM $table"
+    echo "Purged default-added data from: $table"
+done
 
 # Don't forget to reset sequences
 sqlite3 $output_db "VACUUM";
 
-# 4. Load the data, getting the contenttypes table in first
+echo "Purged data that was automatically added via Django/Wagtail migrations"
+
+# Load the data, getting the contenttypes table in first
 PROD_DETAILS_STORAGE=product_details.storage.PDFileStorage \
-    DATABASE_URL=sqlite://$output_db \
     python manage.py loaddata \
         "/tmp/export_contenttypes.json" \
         "/tmp/export_wagtail_locales.json" \
+        "/tmp/export_remainder.json" \
         || all_well=false
 
-check_status 97
+check_status_and_handle_failure "Loading data from JSON"
+
+# There are things we can't omit or redact in the steps above, so
+# we need to manually delete them once we've served their purpose
+echo "Preparing statements for nullifying columns in temporary sql file. (Output is hidden because it's captured from stdout)."
+
+# Convert the array into a comma-separated string suitable for the SQL IN clause
+tables_list=$(printf "'%s'," "${tables_to_wipe_after_import[@]}")
+tables_list=${tables_list%,}  # Remove the trailing comma
+
+# Execute the SQLite commands
+sqlite3 "$output_db" <<EOF
+BEGIN;
+
+-- Step 1: Create a temporary table to store the relations
+CREATE TEMPORARY TABLE fk_relations AS
+SELECT
+    fk_table.name AS table_name,
+    fk."from" AS fk_column
+FROM
+    sqlite_master AS fk_table
+JOIN
+    pragma_foreign_key_list(fk_table.name) AS fk
+WHERE
+    fk."table" IN ($tables_list)
+AND
+    fk_table.type = 'table';
```

Comment:
😱

Reply:
There may be better ways. I admit I did lean on our Robot Collaborators for this part